inst/include/dust/dust.hpp

```diff
@@ -567,41 +556,63 @@ class Dust {
   dust::filter_state<real_t> filter_state_;
 
   void initialise(const pars_t& pars, const size_t step,
-                  const size_t n_particles) {
-    _particles.clear();
-    _particles.reserve(n_particles);
-    for (size_t i = 0; i < n_particles; ++i) {
-      _particles.push_back(Particle<T>(pars, step));
+                  const size_t n_particles, bool set_state) {
+    const size_t n = _particles.size() == 0 ? 0 : n_state_full();
+    Particle<T> p(pars, step);
+    if (n > 0 && p.size() != n) {
+      std::stringstream msg;
+      msg << "'pars' created inconsistent state size: " <<
+        "expected length " << n << " but created length " <<
+        p.size();
+      throw std::invalid_argument(msg.str());
+    }
+    if (_particles.size() == n_particles) {
+#ifdef _OPENMP
+      #pragma omp parallel for schedule(static) num_threads(_n_threads)
+#endif
+      for (size_t i = 0; i < n_particles; ++i) {
+        _particles[i].set_pars(p, set_state);
+      }
```

Comment:
This branch is to save on allocations if np isn't changing?

Reply:
yes, it also lets us do this in parallel (the previous branch can't be done in parallel because we use push_back and we can't resize first because the particles aren't default-constructible)